js/utils/mini-lz4.js

```diff
@@ -0,0 +1,118 @@
+function uncompress(source, uncompressedSize) {
+/*
+Source https://github.com/ethereum/solidity/blob/develop/scripts/ci/mini-lz4.js
```

Comment:
@xermicus This file is taken from Solidity. Is the licensing compliant for use?

Reply:
I don't think so as `ethereum/solidity` appears to be GPL licensed.

Reply:
Ok, so I will rewrite this small part, and we will simply base it on the original mini-lz4. Does that work for you?

Reply:
Sounds good, however I have a bunch of higher level questions related to this:
1. Why is this needed?
2. Why do they use a custom compression implementation (or is it also a custom algorithm)?
3. Why not just use a library?

Reply:
1. It is used for revive wasm compression.
2. it standard lz4 compression, but needs to embeded it in the resolc.js to decompress wasm on the fly
3. What do you mean? Do you want to embed npm lz4 package to resolc.js?

Reply:
Alternative solution: https://github.com/paritytech/revive/pull/176